### DISCUSSION TITLE
[DynamoDB] key_condition_expression inside query.

### DISCUSSION THREAD

**Discussion by Y2KForever** [awslabs/aws-sdk-rust#219] [Q&A] [answered]
Hi, having some issues with `key_condition_expression` giving me the following error: 
`One or more parameter values were invalid: Condition parameter type does not match schema type`.

My current code in Rust looks like this:
```
let table_list = client.list_tables().send().await?;
let names = table_list.table_names;
let table = names.as_ref().unwrap().first().unwrap();

client
      .query()
      .table_name(table)
      .index_name("streamer_id-index")
      .projection_expression("user_id")
      .expression_attribute_names("#sid", "streamer_id")
      .expression_attribute_values(":streamer_id", AttributeValue::S(id.to_string()))
      .key_condition_expression("#sid = :streamer_id") /* Condition parameter type does not match schema type */
      .select(Select::SpecificAttributes)
      .send()
      .await
```

I know this is not the same, but I'm trying to replicate similar code that I have in Node, which is working as intended.

```
const params = {
    TableName: table,
    IndexName: "streamer_id-index",
    ProjectionExpression: "user_id",
    KeyConditionExpression: "#sid = :streamer_id",
    ExpressionAttributeNames: {
      "#sid": "streamer_id",
    },
    ExpressionAttributeValues: {
      ":streamer_id": { N: req.body.id.toString() },
    }
}
  ```
  
  Is key_condition_expression not currently working, or have I missed something? Thought I went over the documentation correctly.

**Comment by rcoh** [comment, marked as answer]
In Rust you used `AttributeValue::S` but in node you used `AttributeValue::N`

**Reply by Y2KForever** [reply]
> 
> 
> In Rust you used `AttributeValue::S` but in node you used `AttributeValue::N`

Christ, I'm blind, thank you!
